staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -411,6 +412,17 @@ func chooseHostInterfaceFromRoute(routes []Route, nw networkInterfacer, addressF
 				klog.V(4).Infof("Found active IP %v ", finalIP)
 				return finalIP, nil
 			}
+			// In case of network setups where default routes are present, but network
+			// interfaces use only link-local addresses (e.g. as described in RFC5549).
+			// the global IP is assigned to the loopback interface
+			loopbackIP, err := getIPFromInterface(LoopbackInterfaceName, family, nw)
```

Comment:
I'm not super excited about using this hardcoded name.  Would it be better to get all the interfaces from `nw` and look for any that have FlagLoopback set?

Reply:
good point, AFAIK this is by convention in Linux , let me change it